crates/pipeline-manager/src/runner/local_runner.rs

```diff
@@ -275,11 +292,76 @@ impl LocalRunner {
 
             // Return the log follow receiver such that it can be re-used when
             // the pipeline is restarted
-            log_follow_request_receiver
+            (log_follow_request_receiver, log_deployment_check_receiver)
         });
         debug!("Logging is operational for pipeline {}", pipeline_id);
         (terminate_sender, join_handle)
     }
+
+    /// Checks the process and working directory of the deployment.
+    /// Returns the fatal error if it is encountered.
+    async fn inner_check(&mut self) -> Result<(), String> {
+        // Pipeline process status must be checkable and not have exited
+        let process_check = if let Some(p) = &mut self.process {
+            match p.try_wait() {
+                Ok(status) => {
+                    if let Some(status) = status {
+                        // If there is an exit status, the process has exited
+                        Err(format!("Pipeline process exited prematurely with {status}"))
+                    } else {
+                        Ok(())
+                    }
+                }
+                Err(e) => {
+                    // Unable to check if it has an exit status, which indicates something went wrong
+                    Err(format!(
+                        "Pipeline process status could not be checked due to: {e}"
+                    ))
+                }
+            }
+        } else {
+            // No process handle
+            Err("The pipeline-manager and this pipeline were previously terminated. Shutdown and start the pipeline again.".to_string())
+        };
+
+        // Pipeline working directory must exist
+        let pipeline_dir = &self.config.pipeline_dir(self.pipeline_id);
+        let working_dir_check = if Path::new(pipeline_dir).is_dir() {
+            Ok(())
+        } else {
+            Err(format!(
+                "Working directory '{}' no longer exists.",
+                pipeline_dir.to_string_lossy()
+            ))
+        };
+
+        // Collect errors
+        let mut errors = vec![];
+        if let Err(e) = &working_dir_check {
+            errors.push(e.clone());
+        }
+        if let Err(e) = &process_check {
+            errors.push(e.clone());
+        }
+
+        // Write to both runner and pipeline log
+        let timestamp = chrono::Utc::now();
+        let timestamp = format!("{}", timestamp.format("%Y-%m-%d %H:%M:%S"));
```

Comment:
```
        let timestamp = format!("{}", timestamp.format("%Y-%m-%d %H:%M:%S"));
```
can we standardize on [utf3899](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.to_rfc3339) or similar for these formats?

Reply:
Yes, sounds good, will make the change

Reply:
I've settled on:
```
let timestamp = chrono::Utc::now();
timestamp.to_rfc3339_opts(SecondsFormat::Micros, true)
```
... which is in line what is currently being printed by `tracing` in the pipeline itself.